### ISSUE TITLE
Windows 10: the linter binary is removed from %PATH% directory upon every vscode update

### ISSUE THREAD

**Issue by AlexCannonball** [plexsystems/vscode-protolint#30]
Hello,

I've found no way to configure a custom `protolint` directory on Windows 10.

I put the binary in `%PATH%` directory (`.../vs code/bin`), but every vscode update removes the binary.

My ideas for fixing this:
- If the directory for protolint binary on Windows is already customizable via preferences.json (or in other way), add these details in readme.md.
- Add some UI in the extension to configure the abovementioned.
- Include the protolint binary in the extension, preventing it from being wiped by vscode updater.

Thank you, and have a great day.

**Comment by jpreese** [comment]
The path of the `protolint` binary can be set by filling in the `protolint.path` configuration value. I'll update the documentation to reflect this.

ref: https://github.com/plexsystems/vscode-protolint/blob/master/package.json#L44

**Comment by jpreese** [comment]
doc ref: https://github.com/plexsystems/vscode-protolint/commit/99c1ea8a262b68f83fcb4b76c48a6144bafbc44e

**Comment by ZCannonball** [comment]
@jpreese Thank you for the reply.
My protolint.exe binary is at `D:\protolint` directory.

I tried inserting it into VS code `settings.json`, but it didn't help:
```json
{
    "protolint.path": "D:\\protolint"
}
```

Then I checked https://github.com/plexsystems/vscode-protolint/blob/master/package.json#L44 and it seems there's a bug there. According to https://code.visualstudio.com/api/references/contribution-points#contributes.configuration the `configuration` must be within `contributes` object, and some other fields don't match the schema. I fixed `package.json` locally and the setting appeared in the UI:
```json
	"contributes": {
		"configuration": {
			"title": "protolint",
			"properties": {
				"protolint.path": {
					"type": "string",
					"default": "$PATH",
					"markdownDescription": "Path of the protolint executable. Defaults to `$PATH`."
				}
			}
		},
		"commands": [
			{
				"command": "protolint.lint",
				"title": "Protolint: Lint protobuf file"
			}
		]
	},
```
However, it didn't help, I'm still gettin the error `protolint was not detected. Download from: https://github.com/yoheimuta/protolint`. I've event tried setting `default` value to `D:\protolint`, nothing changed.

It seems that even after fixing the package.json the extension code just ignores the `protolint.path` value. Could you inspect this further?

**Comment by jpreese** [comment]
I was able to reproduce the issue with the setting not visually appear in the UI. You are right, it was in the wrong section.

However, I was still able to set this value and have the extension use it for the path to `protolint`. That logic is here: https://github.com/plexsystems/vscode-protolint/blob/master/src/linter.ts#L41

Are you sure you're setting the correct `settings.json`? I believe there is a user one and a workspace one. I've verified the user one, and it always seems to pick it up.

**Comment by jpreese** [comment]
> I guess on Windows I can't just open a terminal and call `protolint version` from anywhere. It works only if I set the proper directory in terminal, where the binary exists. So if I don't have the binary in `/bin/` the operation `cp.spawnSync('protolint', ['version'])` just fails. Maybe it's because the downloaded binary is not registered as the console comand, unlike Unix or MacOS.

To do this, you need to place the binary in a folder that is defined on your system path, or add a new folder. ref: https://www.mathworks.com/matlabcentral/answers/94933-how-do-i-edit-my-system-path-in-windows

This should allow you to run `protolint` from anywhere on the command line.

> I've tried debugging the TS extension files locally, however I don't know TS well, so haven't find any clue yet. If you tell me what debug information you need, I can share it.

When you set a path in `protolint.lint` what is the exact error that you see in the bottom right?

**Comment by ZCannonball** [comment]
> When you set a path in `protolint.lint` what is the exact error that you see in the bottom right?
Steps:
- The only protolint binary in the system is `D:\protolint\protolint.exe`.
- User's settings.json contains `"protolint.path": "D:\\protolint\\protolint.exe"`
- That's what I've got in the bottom right:
![image](https://user-images.githubusercontent.com/42497203/229113169-3c51e777-01c2-42a0-915d-e1515b9169b9.png)
As a side notes
- the command `protolint.lint` doesn't work as well and returns the error
![image](https://user-images.githubusercontent.com/42497203/229114087-307e4fc2-ca32-4a3e-a767-418f4dcf3165.png)
- I've locally set this code line https://github.com/plexsystems/vscode-protolint/blob/master/src/extension.ts#L11 to `const result = cp.spawnSync('D:\\protolint\\protolint.exe', ['version']);` and the linter started working with no error messages, so it may show a workaround for Windows.

> To do this, you need to place the binary in a folder that is defined on your system path, or add a new folder

Thank you, I've thought about, most likely it should help. However, it's not obvious when you install the extension, so I would at least reflect it in readme.

**Comment by jpreese** [comment]
@ZCannonball this should fix the issue where protolint was set in `protolint.path` and you were still getting those errors: https://github.com/plexsystems/vscode-protolint/commit/c12685966c34e95bc90ba23684204f5fe215a9a1

`protolint` was always hardcoded when running the version command, rather than using the setting.

**Comment by jpreese** [comment]
Fixed in `v0.8.0`

**Comment by AlexCannonball** [comment]
@jpreese hello, I've checked `0.8.0` (both via VSIX and marketplace) and the issue reproduced. I've checked the compiled `.../out/src/extension.js` and it contains the previous hardcoded path
![image](https://github.com/plexsystems/vscode-protolint/assets/42497203/26195e64-f33d-4a7b-b337-bc874c3761c7)
I guess the tag 0.8.0 doesn't have this fix: https://github.com/plexsystems/vscode-protolint/blob/v0.8.0/src/extension.ts#L11
